### PR TITLE
[master] Moving tests/integration/modules/test_cmdmod.py to pytest.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
moving tests/integration/modules/test_cmdmod.py to pytest.

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Comment by dwoz** [comment]
Re-opening. I'll move this across the finish line for 3008.

**Review by twangboy — APPROVED** [review]

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+def available_python_executable():
+    yield salt.utils.path.which_bin(["python", "python2", "python2.6", "python2.7"])
+
+
+@pytest.fixture
+def runas_usr():
+    runas_usr = "nobody"
+    if salt.utils.platform.is_darwin():
+        runas_usr = "macsalttest"
+    yield runas_usr
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:36]:
```suggestion
@pytest.fixture
def runas_usr():
    yield runas_usr
        with pytest.helpers.create_account() as account:
            yield account.username
    else:
        yield "nobody"
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+
+@pytest.fixture(scope="module")
+def usermod(modules):
+    return modules.user
+
+
+@pytest.fixture(scope="module")
+def available_python_executable():
+    yield salt.utils.path.which_bin(["python", "python2", "python2.6", "python2.7"])
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:28]:
```suggestion
    yield salt.utils.path.which_bin(["python", "python3"])
```
Let's stop searching for python 2

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+from contextlib import contextmanager
+
+import pytest
+import salt.config
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.user
+from tests.support.helpers import SKIP_INITIAL_PHOTONOS_FAILURES, dedent
+
+pytestmark = [pytest.mark.windows_whitelisted]
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:13]:
```suggestion
pytestmark = [
    pytest.mark.windows_whitelisted,
]
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+    if name in usermod.info(name).values():
+        # User already exists; don't touch
+        yield
+    else:
+        # Need to create user for test
+        usermod.add(name)
+        try:
+            yield
+        finally:
+            usermod.delete(name, remove=True)
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:88]:
```suggestion
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+    """
+    cmd = "pwd"
+    tmp_cwd = str(tmp_path)
+    os.chmod(tmp_cwd, 0o711)
+
+    cwd_normal = cmdmod.run_stdout(cmd, cwd=tmp_cwd).rstrip("\n")
+    assert tmp_cwd == cwd_normal
+
+    with _ensure_user_exists(runas_usr, usermod):
+        cwd_runas = cmdmod.run_stdout(cmd, cwd=tmp_cwd, runas=runas_usr).rstrip("\n")
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:424]:
```suggestion
    cwd_runas = cmdmod.run_stdout(cmd, cwd=tmp_cwd, runas=runas_usr).rstrip("\n")
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+@pytest.mark.skip_if_not_root
+@pytest.mark.skip_unless_on_darwin(reason="Applicable to MacOS only")
+@pytest.mark.slow_test
+def test_runas_env(cmdmod, usermod, runas_usr):
+    """
+    cmd.run should be able to change working directory correctly, whether
+    or not runas is in use.
+    """
+    with _ensure_user_exists(runas_usr, usermod):
+        user_path = cmdmod.run_stdout('printf %s "$PATH"', runas=runas_usr)
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:438]:
```suggestion
    user_path = cmdmod.run_stdout('printf %s "$PATH"', runas=runas_usr)
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+    """
+    tmp_cwd = str(tmp_path)
+    os.chmod(tmp_cwd, 0o700)
+
+    with _ensure_user_exists(runas_usr, usermod):
+        cmd_result = cmdmod.run_all(
+            'pwd; pwd; : $(echo "You have failed the test" >&2)',
+            cwd=tmp_cwd,
+            runas=runas_usr,
+        )
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:468]:
```suggestion
    cmd_result = cmdmod.run_all(
        'pwd; pwd; : $(echo "You have failed the test" >&2)',
        cwd=tmp_cwd,
        runas=runas_usr,
    )
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+@pytest.mark.skip_on_windows
+@pytest.mark.skip_if_not_root
+@pytest.mark.destructive_test
+@pytest.mark.slow_test
+def test_runas(cmdmod, usermod, runas_usr):
+    """
+    Ensure that the env is the runas user's
+    """
+    with _ensure_user_exists(runas_usr, usermod):
+        out = cmdmod.run("env", runas=runas_usr).splitlines()
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:485]:
```suggestion
    out = cmdmod.run("env", runas=runas_usr).splitlines()
```

File: tests/pytests/functional/modules/test_cmdmod.py [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+        user = running_username
+    else:
+        user = salt.utils.user.get_specific_user()
+    if user.startswith("sudo_"):
+        user = user.replace("sudo_", "")
+    cmd = cmdmod.run("whoami")
+    assert user.lower() == cmd.lower()
+
+
+@pytest.mark.skip_unless_on_windows(reason="Minion is not Windows")
```
Comment (s0undt3ch) [inline tests/pytests/functional/modules/test_cmdmod.py:522]:
```suggestion
@pytest.mark.skip_unless_on_windows
```
No need to customize message, does not add anything.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
@dwoz added some review comments.

**Comment by dmurphy18** [comment]
@dwoz Have https://github.com/saltstack/salt/pull/65679 actively moving this forward

**Comment by dwoz** [comment]
This will be closed when #65679 passes.

**Review by twangboy — APPROVED** [review]

**Comment by dmurphy18** [comment]
Closing since associated fixing PR https://github.com/saltstack/salt/pull/65679 is merged

**Pull request closed without merging**